my-app/src/components/EvolutionsList.tsx

```diff
@@ -0,0 +1,81 @@
+import { FC, useCallback, useEffect, useState } from 'react';
+import { PokemonType, EvolutionChain, EvolutionsListProps } from '../types';
+import './EvolutionsList.scss';
+
+export const EvolutionList: FC<EvolutionsListProps> = ({ pokemonSpeciesURL }) => {
+  const [evolutions, setEvolutions] = useState<EvolutionChain[]>();
+
+  const getArrayFromRecursiveObject = (obj: EvolutionChain): Array<EvolutionChain> => {
+    const arr: Array<EvolutionChain> = [];
+
+    let currentEvolution: EvolutionChain = obj;
+
+    while (currentEvolution.evolves_to.length && 'evolves_to' in currentEvolution.evolves_to[0]) {
+      arr.push({
+        ...currentEvolution,
+        evolves_to: [],
+      });
+      currentEvolution = currentEvolution.evolves_to[0];
+    }
+
+    arr.push({
+      ...currentEvolution,
+      evolves_to: [],
+    });
+
+    return arr;
+  };
+
+  const fetchPokemonSpriteByName = async (name: string): Promise<string> => {
+    let url = '';
+
+    try {
+      const resp = await fetch(`https://pokeapi.co/api/v2/pokemon/${name}`);
+      const {
+        sprites: { front_default },
+      } = (await resp.json()) as PokemonType;
+      url = front_default;
+    } catch (err) {
+      console.error(err);
+    }
+
+    return url;
+  };
+
+  const fetchEvolutions = useCallback(async () => {
+    const speciesResponse = await fetch(pokemonSpeciesURL);
+    const {
+      evolution_chain: { url },
+    } = (await speciesResponse.json()) as { evolution_chain: { url: string } };
+
+    const evolutionResponse = await fetch(url);
+    const { chain } = (await evolutionResponse.json()) as { chain: EvolutionChain };
```

Comment:
Use [type guard](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) to cast types.